do

```diff
@@ -15,211 +15,450 @@ from epimodel.gleam import Batch, batch
+@click.pass_context
+def export_gleam_batch(ctx, batch_file, out_dir, overwrite):
+    """
+    Export batch of definitions for GLEAM.
+
+    By default exports to 'gleamviz_sims_dir' as specified in config.yml.
+
+    BATCH_FILE: The batch-*.hdf5 file with batch spec to be updated.
+    """
+    batch = Batch.open(batch_file)
+    gdir = ctx.obj["CONFIG"]["gleamviz_sims_dir"]
+    if out_dir is not None:
+        gdir = out_dir
+    log.info(f"Creating GLEAM XML definitions for batch {batch_file} in dir {gdir} ...")
     batch.export_definitions_to_gleam(
-        Path(gdir).expanduser(), overwrite=args.overwrite, info_level=logging.INFO
+        Path(gdir).expanduser(), overwrite=overwrite, info_level=logging.INFO
     )
 
 
-def create_parser():
-    ap = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-    ap.add_argument("-d", "--debug", action="store_true", help="Debugging logs.")
-    ap.add_argument("-C", "--config", default="config.yaml", help="Config file.")
-    sp = ap.add_subparsers(title="subcommands", required=True, dest="cmd")
-
-    upp = sp.add_parser(
-        "update_johns_hopkins", help="Fetch data from Johns Hopkins CSSE."
+@action.command()
+@click.argument("batch_file", type=click.Path(exists=True))
+@click.argument("estimates", type=click.Path(exists=True))
+@click.option("-c", "--comment", type=str, help="A short comment (to be part of path).")
+@click.option(
+    "-p", "--pretty-print", is_flag=True, help="Pretty-print exported JSON files."
+)
+@click.pass_context
+def web_export(ctx, batch_file, estimates, comment, pretty_print):
+    """
+    Create data export for web.
+
+    Saved in the directory specified by 'output_dir' in config.yml.
+
+    BATCH_FILE: A result HDF file of import-gleam-batch step
+
+    ESTIMATES: CSV file containing the current estimates
+    """
+    process_export(
+        ctx.obj["CONFIG"],
+        ctx.obj["RDS"],
+        ctx.obj["DEBUG"],
+        comment,
+        batch_file,
+        estimates,
+        pretty_print,
     )
-    upp.set_defaults(func=update_johns_hopkins)
 
-    upf = sp.add_parser("update_foretold", help="Fetch data from Foretold.")
-    upf.set_defaults(func=update_foretold)
 
-    ibp = sp.add_parser("import_gleam_batch", help="Load batch results from GLEAM.")
-    ibp.add_argument(
-        "BATCH_FILE", help="The batch-*.hdf5 file with batch spec to be updated."
-    )
-    ibp.add_argument(
-        "-M", "--allow-missing", action="store_true", help="Skip missing sim results."
-    )
-    ibp.add_argument(
-        "--overwrite",
-        action="store_true",
-        help="Overwrite existing `new_fraction` imported table.",
+@action.command()
+@click.option(
+    "-d",
+    "--dir",
+    "dir_",
+    type=click.Path(exists=True),
+    help="The generated export directory to upload from.",
+)
+@click.option(
+    "-c",
+    "--channel",
+    type=str,
+    default="staging",
```

Comment:
Could we make this argument mandatory i.e. remove the default value?

Reply:
Sure (I made it optional because that was how it was defined in the original file) -- out of interest could I ask your reasoning behind doing this?

Reply:
1. The user should always consciously pick a channel to push to.
2. I don't see why `staging` should be privileged, if anything it should be `testing` imho.

Reply:
Ah, that makes sense. (For the docstring, are the only channels people might want to push to `main`, `staging` and `testing`?)

Reply:
The channel field is also used for 'custom sites' such as the one for Pakistan (I believe they used `balochistan`). But for the docstring putting those and maybe a mention of custom sites is ok!

Reply:
I am also fine with having this mandatory.

Reply:
Made the change :) I've made it into an argument.